NBitcoin/Utils.cs

```diff
@@ -739,21 +739,25 @@ public static void Shuffle<T>(T[] arr, Random rand)
 				arr[fromIndex] = to;
 			}
 		}
-		public static void Shuffle<T>(List<T> arr, Random rand)
+		public static void Shuffle<T>(List<T> arr, int start, Random rand)
 		{
 			rand = rand ?? new Random();
-			for (int i = 0; i < arr.Count; i++)
+			for (int i = start; i < arr.Count; i++)
 			{
-				var fromIndex = rand.Next(arr.Count);
+				var fromIndex = rand.Next(start, arr.Count - start);
 				var from = arr[fromIndex];
 
-				var toIndex = rand.Next(arr.Count);
+				var toIndex = rand.Next(start, arr.Count - start);

```

Comment:
This works okay but you could use some syntax sugar, something from:
 
```c#
var segment = arr[start..]; 

var fromIdx = rand.TakeOne(segment); 
var toIdx = rand.TakeOne(segment); 
(segment[fromIdx], segment[toIdx]) = (segment[toIdx], segment[fromIdx]);
```

Reply:
I must admit that I have hard time to understand what it does :p

Reply:
ha I agree I am not sure I would call this sugar `arr[start..]`